snmp/datadog_checks/snmp/snmp.py

```diff
@@ -42,6 +43,56 @@ def reply_invalid(oid):
     return noSuchInstance.isSameTypeWith(oid) or noSuchObject.isSameTypeWith(oid)
 
 
+def discover_instances(config, interval, check_ref):
```

Comment:
Nit: would it make sense to take the opportunity to move this to a dedicated `discovery.py` module? This big function with lots of low-level details living just above the check class looks a bit off.